pkg/kubelet/cm/cgroup_v2_manager_linux.go

```diff
@@ -144,35 +128,6 @@ func (c *cgroupV2impl) getCgroupCPUConfig(cgroupPath string) (*ResourceConfig, e
 	return &ResourceConfig{CPUShares: &cpuShares, CPUQuota: &cpuLimit, CPUPeriod: &cpuPeriod}, nil
 }
 
-func (c *cgroupV2impl) setCgroupCPUConfig(cgroupPath string, resourceConfig *ResourceConfig) error {
-	if resourceConfig.CPUQuota != nil {
-		if resourceConfig.CPUPeriod == nil {
-			return fmt.Errorf("CpuPeriod must be specified in order to set CpuLimit")
-		}
-		cpuLimitStr := Cgroup2MaxCpuLimit
-		if *resourceConfig.CPUQuota > -1 {
-			cpuLimitStr = strconv.FormatInt(*resourceConfig.CPUQuota, 10)
-		}
-		cpuPeriodStr := strconv.FormatUint(*resourceConfig.CPUPeriod, 10)
-		cpuMaxStr := fmt.Sprintf("%s %s", cpuLimitStr, cpuPeriodStr)
-		if err := os.WriteFile(filepath.Join(cgroupPath, "cpu.max"), []byte(cpuMaxStr), 0700); err != nil {
-			return fmt.Errorf("failed to write %v to %v: %w", cpuMaxStr, cgroupPath, err)
-		}
-	}
-	if resourceConfig.CPUShares != nil {
-		cpuWeight := cpuSharesToCPUWeight(*resourceConfig.CPUShares)
```

Comment:
I think we need to preserve this conversion. I think this can just be done in the common path, so both CPU shares & CPU weight are set, and libcontainer just uses the correct one depending on if it's cgroups v1 or v2

https://github.com/kubernetes/kubernetes/blob/1bbe775d5ffb131636193fe0bc15a8fcc0cd6fd6/vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2/cpu.go#L22

Reply:
Hey @tallclair!
I believe this conversion already takes place in the current state of this PR.

The `setCgroupCPUConfig` is now unified here: https://github.com/kubernetes/kubernetes/pull/124216/files#diff-2f332b11d91d53583a6eeca75fc6e124aee6bc547a264e6987b89b1c786e0674R257.

In turn, the `Update()` function uses `libctCgroupConfig()` to convert the cgroup config to a runc struct:
https://github.com/kubernetes/kubernetes/blob/b845137026be136ec12d798a2b729ce8b99b85af/pkg/kubelet/cm/cgroup_manager_linux.go#L358-L364

And indeed this function handles the conversion between cpu shares to cpu weight:
https://github.com/kubernetes/kubernetes/blob/b845137026be136ec12d798a2b729ce8b99b85af/pkg/kubelet/cm/cgroup_manager_linux.go#L199-L204
https://github.com/kubernetes/kubernetes/blob/b845137026be136ec12d798a2b729ce8b99b85af/pkg/kubelet/cm/cgroup_manager_linux.go#L274-L291
https://github.com/kubernetes/kubernetes/blob/b845137026be136ec12d798a2b729ce8b99b85af/pkg/kubelet/cm/cgroup_manager_linux.go#L259-L267

WDYT?

Reply:
Ah, thanks. I missed that. Looks good then.